src/constructs/autoscaling/asg.ts

```diff
@@ -80,6 +80,23 @@ function wireStageDependentProps(stack: GuStack, stageDependentProps: GuStageDep
   };
 }
 
+/**
+ * Construct which creates an Auto Scaling group.
+ *
+ * By default, all EC2 instances in this group will use [[`GuInstanceRole`]],
+ * which provides common permissions (e.g. the ability to download an artifact and write logs to our central ELK stack).
+ *
+ * If additional IAM permissions are required, a custom role can be provided via the `role` prop.
+ * You may wish to instantiate [[`GuInstanceRole`]] yourself as a basis for this custom role, as it allows custom permissions
```

Comment:
Not for this PR, but the ASG (or pattern) could take this prop and pass it down for ease.

Reply:
Yep, agreed; sometimes writing the docs is a good way to make yourself think "we could make this easier!" 😀

Reply:
cc @stephengeller, as I know you're currently looking at overriding defaults via the pattern